packages/hdwallet-ledger/src/ledger.ts

```diff
@@ -327,7 +327,20 @@ export class LedgerHDWallet implements core.HDWallet, core.BTCWallet, core.ETHWa
   }
 
   public async getFirmwareVersion (): Promise<string> {
-    const { version } = await this.getFeatures()
+    await this.transport.open()
+
+    var version
+    try {
+        var { version } = await this.getFeatures()
+    } catch(err) {
+        if (err.message.includes('0x6d00'))
+            throw new core.WrongApp('Ledger', 'Dashboard')
+
+        console.error(err)
+    }
+
+    await this.transport.close()
```

Comment:
do we need to `close()` from a `finally` clause?

Reply:
Good catch, thanks!